gen/builders/builder_message.go

```diff
@@ -112,19 +112,24 @@ func (b *MessageVectorBuilder) CommitApplies() {
 			Bytes: MustSerialize(am.Message),
 			Epoch: &epoch,
 		})
-		b.vector.Post.Receipts = append(b.vector.Post.Receipts, &schema.Receipt{
-			ExitCode:    int64(am.Result.ExitCode),
-			ReturnValue: am.Result.Return,
-			GasUsed:     am.Result.GasUsed,
-		})
+
+		// am.Result may still be nil if the message failed to be applied
+		if am.Result != nil {
```

Comment:
Maybe a more idiomatic way of doing this would be:

```go
var receipt *schema.Receipt
if am.Result != nil {
    receipt = &schema.Receipt{
        ExitCode:    int64(am.Result.ExitCode),
        ReturnValue: am.Result.Return,
        GasUsed:     am.Result.GasUsed,
    }
}
b.vector.Post.Receipts = append(b.vector.Post.Receipts, receipt)
```